mono/mini/mini-runtime.c

```diff
@@ -4691,16 +4693,30 @@ register_icalls (void)
 
 MonoJitStats mono_jit_stats = {0};
 
+void
+mono_stats_lock (void)
+{
+	mono_coop_mutex_lock (&stats_lock);
+}
+
+void
+mono_stats_unlock (void)
+{
+	mono_coop_mutex_unlock (&stats_lock);
+}
+
 /**
  * Counters of mono_stats and mono_jit_stats can be read without locking during shutdown.
- * For all other contexts, assumes that the domain lock is held.
+ * Mono stats are locked/unlocked in this function call.
+ * Use mono_stats_lock and mono_stats_unlock when reading/writing stats in other contexts.
```

Comment:
This comment should go over by the declaration of `mono_stats` rather than here.